src/linear_algebra/basic.lean

```diff
@@ -1474,6 +1474,38 @@ end field
 
 end linear_map
 
+section
+variables (R) [ring R] (X : Type*) (M) [add_comm_group M] [module R M]
+
+/-- The bijection underlying the free-forgetful adjunction for R-modules. -/
+noncomputable def finsupp.hom_equiv : ((X →₀ R) →ₗ[R] M) ≃ (X → M) :=
```

Comment:
This looks quite similar to `finsupp.lsum.symm.trans _` with `_` as `(X → (R →ₗ[R] M)) ≃ (X → M)`. Is it easier to build things that way?

Reply:
To wit:

```lean
noncomputable def finsupp.hom_equiv : ((X →₀ R) →ₗ[R] M) ≃ (X → M) :=
(finsupp.lsum : _ ≃+ _).to_equiv.symm.trans $ equiv.arrow_congr (equiv.refl _)
{ to_fun := λ f, f 1,
  inv_fun := λ m, { map_smul' := λ r r', by simp [mul_smul], ..(smul_add_hom R M).flip m },
  left_inv := λ x, by { ext, simp },
  right_inv := λ x, by { simp } }
```

Reply:
Thanks, that is much better!